### PR TITLE
UI for TLS hardening

### PR DESCRIPTION
**Pull request by stephdl**
https://github.com/NethServer/dev/issues/5438

UI to harden the TLS protocol

### CODE REVIEW

File: root/usr/share/nethesis/NethServer/Template/TlsPolicy.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+echo $view->header()->setAttribute('template', $T('TlsPolicy_Description'));
+
+echo $view->panel()
+    ->insert($view->selector('policy', $view::SELECTOR_DROPDOWN))
+    ->insert($view->fieldset()->setAttribute('template', $T('ServiceExemption_label')))
+    ->insert($view->checkbox('HttpdAdminExeptionStatus','enabled')->setAttribute('uncheckedValue', 'disabled'))
```
Comment (DavidePrincipi) [inline root/usr/share/nethesis/NethServer/Template/TlsPolicy.php]: `exeption`: is it a typo?

Why this checkbox is needed? How is a "service exemption"  defined? How does it differ from the base policy?

Comment (stephdl) [reply root/usr/share/nethesis/NethServer/Template/TlsPolicy.php]: yes probably :D

**Comment by stephdl** [comment]
@DavidePrincipi do you think that every service should have is exemption checkbox to use the legacy cipher (actual cipher)
I did it because you asked for http-d-admin but we can do it

File: root/usr/share/nethesis/NethServer/Module/TlsPolicy.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public function prepareView(\Nethgui\View\ViewInterface $view)
+    {
+        parent::prepareView($view);
+        $view['policyDatasource'] = \Nethgui\Renderer\AbstractRenderer::hashToDatasource(array(
+                'legacy' => $view->translate('LEGACY'),
+        ));
+    }
+    protected function onParametersSaved($changes)
+    {
+        $this->getPlatform()->signalEvent('tls-policy-save');
```
Comment (DavidePrincipi) [inline root/usr/share/nethesis/NethServer/Module/TlsPolicy.php]: Set evento name with " &" suffix to run a detached event (ie ``tls-policy-save &``)

File: root/usr/share/nethesis/NethServer/Help/en/NethServer_Module_TlsPolicy.rst [hunk 3]
```diff
@@ -0,0 +1,7 @@
+========================
+Transport Layer Security
+========================
+
+TLS is a cryptographic protocols that provide communications security over a computer network. Several versions of the protocols find widespread use in applications such as web browsing, email, Internet faxing, instant messaging, and voice over IP (VoIP). The TLS protocol aims primarily to provide privacy and data integrity between two communicating computer applications.
+
+You can enforce the TLS policy, or choose the legacy one if your clients are not supported/maintained anymore (Windows XP for example)
```
Comment (DavidePrincipi) [inline root/usr/share/nethesis/NethServer/Help/en/NethServer_Module_TlsPolicy.rst]: We could apply to new modules a new inline help pattern... I'm pushing this approach

https://github.com/NethServer/nethserver-samba/blob/master/root/usr/share/nethesis/NethServer/Help/en/NethServer_Module_FileServer.rst

A link to a special admin manual section. The main reason is: the manual is translated on Transifex

File: root/usr/share/nethesis/NethServer/Module/TlsPolicy.php (hunk 2 above)
Comment (stephdl) [reply root/usr/share/nethesis/NethServer/Module/TlsPolicy.php]: done

File: root/usr/share/nethesis/NethServer/Help/en/NethServer_Module_TlsPolicy.rst (hunk 3 above)
Comment (stephdl) [reply root/usr/share/nethesis/NethServer/Help/en/NethServer_Module_TlsPolicy.rst]: done, see my PR in the doc side

**Pull request merged**